src/layouts/sidebar-sidecar/components/table-of-contents/index.tsx

```diff
@@ -0,0 +1,51 @@
+import { ReactElement } from 'react'
+import classNames from 'classnames'
+import { useDeviceSize } from 'contexts'
+import getTruncatedTitle from './utils/get-truncated-title'
+import { TableOfContentsHeading, TableOfContentsProps } from './types'
+import { useActiveSection } from './use-active-section'
+import s from './table-of-contents.module.css'
+
+const TABLE_OF_CONTENTS_LABEL_ID = 'table-of-contents-label'
+
+const TableOfContents = ({ headings }: TableOfContentsProps): ReactElement => {
```

Comment:
Curious why you opted to keep this nested within the layout folder? 💭

Reply:
Mainly because the styling of it is currently super tied to being placed in `SidebarSidecarLayout` (the right alignment for example). I'm definitely open to moving it to the main components folder in the future if we generalize it enough for reuse! I think we'd have to strip the styling away and then add some `className` props to enable the current styling from `SidebarSidecarLayout`. But I didn't want its placement in the main components folder to imply "this can be used anywhere!" since it's not ready for reuse at the moment.

Reply:
Makes sense! 👍